src/malli/json_schema.cljc

```diff
@@ -82,25 +82,28 @@
 (defmethod accept :orn [_ _ children _] {:anyOf (map last children)})
 
 (defmethod accept ::m/val [_ _ children _] (first children))
+
 (defmethod accept :map [_ schema children _]
-  (let [required (->> children (filter (m/-comp not :optional second)) (mapv first))
-        additional-properties (:closed (m/properties schema))
+  (let [ks (set (m/explicit-keys schema))
+        default (some->> children (remove (m/-comp ks first)) first last)
```

Comment:
hmm, you can't use `(default-schema)` or `(-default-entry-schema)` here because the children have been walked already?

Reply:
yes